hack/generate-versions.sh

```diff
@@ -15,20 +15,20 @@ TMP_BIN="$(pwd)/tmp/bin"
 # Ensure that we use the binaries from the versions defined in hack/tools/go.mod.
 PATH="${TMP_BIN}:${PATH}"
 
+MAIN_BRANCH="master"
+
 # Set default variable values
 : "${VERSION_FILE:=jsonnet/versions.yaml}"
-: "${INTERACTIVE:=true}"
+# PULL_BASE_REF will also be set by CI
+: "${PULL_BASE_REF:=master}"
 
-version_from_remote() {
-	curl --retry 5 --silent --fail "https://raw.githubusercontent.com/${1}/master/VERSION"
-}
+if [ "$PULL_BASE_REF" != "$MAIN_BRANCH" ]; then
+	echo >&2 "Components versions are only updated on '${MAIN_BRANCH}' branch for now. Nothing to do against '${PULL_BASE_REF}'."
+	exit 0
+fi
 
-# Fallback mechanism when VERSION file is empty or not found
-version_from_user() {
-	ver=""
-	echo >&2 -n "Cannot determine version of ${1}. Please provide version manually (without alphabetical prefixes) and press ENTER: "
-	read -r ver
-	echo "$ver"
+version_from_remote() {
+	curl --retry 5 --silent --fail "https://raw.githubusercontent.com/${1}/${MAIN_BRANCH}/VERSION"
```

Comment:
(we can still use this if we change (I think we should do that for clarity) components versions on release-xxx branches when fixing them)